tensorflow_datasets/core/dataset_info.py

```diff
@@ -407,7 +408,53 @@ def initialize_from_bucket(self):
       out_fname = os.path.join(tmp_dir, os.path.basename(fname))
       gcs_utils.download_gcs_file(fname, out_fname)
     self.read_from_directory(tmp_dir)
-
+  
+  def show_batch(self, ds, max_images=None, rows=None, factor=3):
+    """Shows some random image samples from the image-dataset.
+       Works only with batch data. If image shapes are different then creating
+       batch will raise error. Possibly resize and create the batch data or use
+       batch_size=1.
+    
+    Args:
+      ds: `tf.data.Dataset`, with consistent batch size.
+      max_images: `int`, maximum number of images to show from a batch. 
+        Should be <= batch_size. Defaults to None.
+      rows: `int`, number of rows the display grid should have. Defaults to None.
+      factor: `int`, to control the size of the display grid. Defaults to 3.
+    """
+    show_label = False
+    if 'label' in self._features.keys():
+      show_label = True
+    for data in ds.take(1):
+      X = dataset_utils.as_numpy(data['image'])
+      if show_label:
+        y = dataset_utils.as_numpy(data['label'])
+  
+    if not rows:
+      if not max_images:
+        rows = int(X.shape[0]**0.5)
+      else:
+        rows = int(max_images**0.5)
+      columns = rows  
+    else:  
+      if max_images:
+        columns = min(X.shape[0], max_images) // rows
+      else:
+        columns = X.shape[0] // rows
+  
+    fig = plt.figure(figsize=(factor*columns, factor*rows))
+    fig.subplots_adjust(hspace=3*1/factor, wspace=1/factor)
+
+    for i in range(rows*columns):
+      fig.add_subplot(rows, columns, i+1)
+      img = X[i]
+      if img.shape[2] == 1:
+        img = img.reshape(img.shape[:2])
+      plt.imshow(img, cmap='gray')
+      if show_label:
+        plt.xlabel("label: int={}, str={}".format(y[i], self._features['label'].int2str(y[i])))
+    plt.show()
+  
```

Comment:
As this file start being bigger and bigger, and this function is probably gonna get more complicated overtime, maybe you could move it to another module, and call it from this:

```
def show_examples(self, *args, **kwargs):
  return show_example_lib.show_examples(*args, **kwargs)
```